syntax = "proto3";

service AriaPeer {
  rpc PerformHandshake(HandshakeRequest) returns (HandshakeResponse);
  rpc SyncMemory(SyncMemoryRequest) returns (SyncMemoryResponse);
}

message HandshakeRequest {
  bytes sender_public_key = 1;
}

message HandshakeResponse {
  bytes peer_public_key = 1;
}

message SyncMemoryRequest {
  string sender_id = 1;
  bytes encrypted_memory = 2;
  bytes signature = 3;
  string current_cycle_id = 4;
  repeated string active_plugins = 5;
}

message SyncMemoryResponse {
  string message = 1;
  string peer_cycle_id = 2;
  repeated string active_plugins = 3;  // âœ… Ensure this is part of response
}